numpy/core/src/multiarray/multiarray_tests.c.src

```diff
@@ -1591,6 +1592,44 @@ get_fpu_mode(PyObject *NPY_UNUSED(self), PyObject *args)
 }
 
 
+static char call_npy_cabs_doc[] = (
+    "npy_cabs(z)\n"
+    "\n"
+    "(double(npy_cabsf(z)), double(npy_cabs(z)),\n"
+    " double(npy_cabsl(z)))");
+
+static PyObject *
+call_npy_cabs(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    Py_complex py_z;
+    npy_cfloat zf;
+    float wf;
+    npy_cdouble z;
+    double w;
+    npy_clongdouble zl;
+    double wl;
+
+    if (!PyArg_ParseTuple(args, "D", &py_z)) {
+        return NULL;
+    }
+
+    zf.real = py_z.real;
+    zf.imag = py_z.imag;
+
+    z.real = py_z.real;
+    z.imag = py_z.imag;
+
+    zl.real = py_z.real;
+    zl.imag = py_z.imag;
+
+    wf = npy_cabsf(zf);
```

Comment:
Scipy test suite finds a few FPU mode changes, but all of them seem to be due to just cabs.